ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestExecutorUtil.java

```diff
@@ -177,28 +177,37 @@ private static Topic buildSinkTopic(
   ) {
     final String kafkaTopicName = sinkDataSource.getKafkaTopicName();
 
-    final Optional<ParsedSchema> schema = getSchema(sinkDataSource, schemaRegistryClient);
+    final Optional<ParsedSchema> keySchema = getSchema(
+        sinkDataSource.getKsqlTopic().getKeyFormat().getFormat(),
+        sinkDataSource.getKsqlTopic().getKafkaTopicName(),
+        schemaRegistryClient
+    );
+    final Optional<ParsedSchema> valueSchema = getSchema(
+        sinkDataSource.getKsqlTopic().getValueFormat().getFormat(),
+        sinkDataSource.getKsqlTopic().getKafkaTopicName(),
+        schemaRegistryClient
+    );
 
-    final Topic sinkTopic = new Topic(kafkaTopicName, schema);
+    final Topic sinkTopic = new Topic(kafkaTopicName, keySchema, valueSchema);
 
     stubKafkaService.ensureTopic(sinkTopic);
     return sinkTopic;
   }
 
   private static Optional<ParsedSchema> getSchema(
-      final DataSource dataSource,
+      final String format,
+      final String topicName,
       final SchemaRegistryClient schemaRegistryClient
   ) {
     final Format valueFormat = FormatFactory
-        .fromName(dataSource.getKsqlTopic().getValueFormat().getFormat());
+        .fromName(format);
 
     if (!valueFormat.supportsFeature(SerdeFeature.SCHEMA_INFERENCE)) {
       return Optional.empty();
     }
 
     try {
-      final String subject =
-          dataSource.getKafkaTopicName() + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX;
+      final String subject = topicName + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX;
```

Comment:
This method is called is called to retrieve the key schema above (line 180) even though the method hardcodes the value suffix. Is key schema inference meant to be supported in QTT / the testing tool in this PR or not? I see in `initializeTopics()` below that only the value schema is checked as well.

Reply:
I started going down the path of supporting QTT tests, but because AVRO key formats aren't yet supported I couldn't get too far with testing. I figured the changes I've made here will be useful in a bit so I just left them in - but it is (as you can tell) a little raw...

I'll clean it up.